src/Stratis.Bitcoin/Utilities/FileStorage.cs

```diff
@@ -45,7 +46,14 @@ public void SaveToFile(T toSave, string fileName, bool saveBackupFile = false)
             string newFilePath = $"{filePath}.{uniqueId}.new";
             string tempFilePath = $"{filePath}.{uniqueId}.temp";
 
-            File.WriteAllText(newFilePath, JsonConvert.SerializeObject(toSave, Formatting.Indented));
+            if (minimalSize)
```

Comment:
I'm thinking we should just always do the size optimization and then if the user so wish, turn "on" formatting etc. I mean, how often will the user look at the wallet file?

@MithrilMan what do you think?

Reply:
well, being this an helper class I'd leave the option to do whatever we want with storage
changing everything to minimalized, would mean changing current stored file for all the classes that use this helper and I don't know if I should do that change in this PR, I think would be better to address case by case

Reply:
Having said this, I'd minimize every file we produce, I just didn't wanted to do that on a single PR in case the format matters somehow, but if you are sure nothing will break up if we change peer, federation wallet, etc... I can just set a default value of `minimalSize` to true and the problem is solved and still if some usecase need to produce a readable file it's possible setting minimalSize to false